### PR TITLE
make `tentez apply` idempotent

### PR DESCRIPTION
**Pull request by FeLvi-zzz**
make `tentez apply` idempotent.

This PR makes you able to skip been used steps.
If a current listener rule weight is old:new = 50:50 and you switch its weight old:new = 0:100, it will be ok.
Although if a current listener rule weight is old:new = 50:50 and you switch its weight old:new = 100:0, it will be skipped.

### CODE REVIEW

File: awsListenerRule.go [hunk 1]
```diff
@@ -19,15 +19,45 @@ type AwsListenerRule struct {
...
+				targetGroupWeightMap[*tgTuple.TargetGroupArn] = float64(*tgTuple.Weight)
+			}
+		}
+		weightSum := 0.0
+		for _, v := range targetGroupWeightMap {
+			weightSum += v
+		}
+
+		oldWeight, ok := targetGroupWeightMap[r.Switch.Old]
+		if !ok || oldWeight/weightSum < float64(targetWeight.Old)/float64(targetWeight.New+targetWeight.Old) {
```
Comment (hatajoe) [inline awsListenerRule.go:51]: [memo] この辺のビジネスロジックが２箇所に散ってるのが若干気になるが、ここだけ関数抽出するのは不自然で、多分 AWS API 呼び出しや TargetGroup を interface 化しないといけなそうなでそれは少し大げさという気もする。もう１箇所出てきたら考えるくらいでええかな。

Comment (FeLvi-zzz) [reply awsListenerRule.go]: refactored https://github.com/FeLvi-zzz/tentez/pull/22/commits/8a9a41b5f269a0d024ba39e7f4baa412b6344268

File: apply.go [hunk 2]
```diff
@@ -78,10 +78,15 @@ func execSwitch(targets map[string]Targets, weight Weight, cfg Config) error {
 			i++
 
 			fmt.Fprintf(cfg.io.out, "%d. %s ", i, target.getName())
-			if err := target.execSwitch(weight, cfg); err != nil {
-				return err
+			if err := target.execSwitch(weight, false, cfg); err != nil {
+				_, ok := err.(SkipSwitchError)
+				if !ok {
+					return err
+				}
```
[the remark this answers is not in the document]
Comment (FeLvi-zzz) [reply apply.go]: it's not errors.Is but errors.As, is it?
fixed https://github.com/FeLvi-zzz/tentez/pull/22/commits/1ce577d3b94abf98a049af2bbf7cacaf8b4c608d.

**Comment by FeLvi-zzz** [comment]
I'll rebase commits to understand easily after reviewed.

**Review by hatajoe — APPROVED** [review]
LGTM

**Comment by FeLvi-zzz** [comment]
Thank you for your reviewing!

**Pull request merged**